File: Editorial/1121_Divide_Array_Into_Increasing_Sequences/C/1_Approach_Maximum_Value.c
bool canDivideIntoSubsequences(int* nums, int numsSize, int k) {
    int pre = nums[0], cnt = 0;
    for (int i = 0; i < numsSize; i++) {
        int n = nums[i];
        if (n == pre) {
            cnt++;
        } else {
            pre = n;
            cnt = 1;
        }
        if (cnt * k > numsSize) {
            return 0;
        }
    }
    return 1;
}

File: Editorial/1101-1200/1121_Divide_Array_Into_Increasing_Sequences/Java/1_Approach_Maximum_Value.java
class Solution {

    public boolean canDivideIntoSubsequences(int[] nums, int k) {
        int pre = nums[0];
        int cnt = 0;
        for (int n : nums) {
            if (n == pre) {
                cnt++;
            } else {
                pre = n;
                cnt = 1;
            }
            if (cnt * k > nums.length) {
                return false;
            }
        }
        return true;
    }
}
